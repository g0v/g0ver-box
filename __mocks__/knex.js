import knex from 'jest-mock-knex';

export { client } from 'jest-mock-knex';

export default knex;

process.setMaxListeners(0);
