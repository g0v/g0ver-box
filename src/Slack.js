import _ from 'lodash';
import slack from 'slack';

export const bot = slack.rtm.client();

const TOKEN = process.env.SLACK_TOKEN;

if (TOKEN) {
  bot.listen({ token: TOKEN });
}

bot.hello((message) => {
  console.log(`Got a message: ${message.type}`);
});

const call = (name, req = {}) => (
  new Promise((resolve, reject) => {
    const fn = _.get(slack, name);
    fn(
      { ...req, token: TOKEN },
      (err, res) => (err ? reject(err) : resolve(res))
    );
  })
);

export default {
  userInfo: async req => await call('users.info', req),
  postMessage: async req => await call('chat.postMessage', { ...req, as_user: true }),
  postMultiMessage: async (users, message) => _.map(users, user => bot.ws.send(JSON.stringify({
    id: `${user}::${Date.now()}`, type: 'message', channel: user, text: message,
  }))),
  channelInfo: async req => await call('channels.info', req),
  channelJoin: async req => await call('channels.join', req),
};
