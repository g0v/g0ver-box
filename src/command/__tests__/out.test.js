import { client } from 'knex';
import Slack from '../../Slack';
import index from '../';

describe('out command', () => {
  const data = { channel: 'D100', name: 'yutin', text: 'out' };

  it('when g0ver no any task', async () => {
    client.mockReturnValueOnce([]);
    await index(data);
    expect(client).toMatchSnapshot();
    expect(Slack.postMessage).toHaveBeenLastCalledWith({ channel: 'D100', text: 'yutin 目前沒有入坑' });
  });

  it('when g0ver have been a task', async () => {
    client.mockReturnValueOnce([{ id: 'XYZ', note: 'g0ver box' }]);
    await index(data);
    expect(client).toMatchSnapshot();
    expect(Slack.postMessage).toHaveBeenLastCalledWith({ channel: 'D100', text: 'yutin 跟大家分享 g0ver box 吧!' });
  });
});
