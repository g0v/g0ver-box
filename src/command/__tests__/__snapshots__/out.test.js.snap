// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`out command when g0ver have been a task 1`] = `
Array [
  "select * from task where expired_at > DATE and deleted_at is null and user_id = U03B2AB13 limit 1",
]
`;

exports[`out command when g0ver no any task 1`] = `
Array [
  "select * from task where expired_at > DATE and deleted_at is null and user_id = U03B2AB13 limit 1",
]
`;
